### PR TITLE
Test memory leaks

### PR DESCRIPTION
**Pull request by gjtorikian**
This is a [libcmark](https://github.com/jgm/cmark) trick I picked up recently. We can build a test that runs Valgrind over the entire test suite. 

In the best case scenario, Valgrind should return an exit status of 0. Otherwise, there's a memory leak somewhere that needs to be plugged.

### CODE REVIEW

**Comment by gjtorikian** [comment]
I can fix the failure (there _is_ a memory leak) if the rest of the PR is desired.

File: .travis.yml [hunk 1]
```diff
@@ -19,4 +19,4 @@ before_install:
       && cd ..
       && rm gtest-1.7.0.zip
 
-script: mkdir build && cd build && cmake .. -Dtest=ON && make && test/runTests
+script: mkdir build && cd build && cmake .. -Dtest=ON && make && test/runTests && make memcheck
```
Comment (swolchok) [inline .travis.yml]: it would be nice if this was a debug build so that the errors were understandable, preexisting problem though

**Comment by swolchok** [comment]
Seems like a fine idea to me. The valgrind output doesn't seem to indicate a memory leak (or, AFAICT without line numbers in the error, an actual problem), so I'm interested to see what the fix is.

**Comment by gjtorikian** [comment]
It turns out that there isn't a memory leak after all. [Erroneous reporting on `strdup`](https://software.intel.com/en-us/articles/valgrind-memcheck-gives-uninitialized-value-false-positives-on-string-routines) is apparently a known thing; I've opted to follow the recommendation and [suppress the error](https://wiki.wxwidgets.org/Valgrind_Suppression_File_Howto).

> without line numbers in the error

I can add this if you want (it's a single flag: `--track-origins=yes`). But, providing line numbers in the Valgrind output greatly slows the run down, so I tend to not include it.

**Comment by swolchok** [comment]
Looking good. Can you please rearrange your stack so that there isn't an intermediate commit that breaks the build?

**Comment by gjtorikian** [comment]
@swolchok Done!

**Pull request merged**